code/addons/interactions/src/components/MethodCall.tsx

```diff
@@ -340,16 +340,32 @@ export const ElementNode = ({
   );
 };
 
-export const DateNode = ({ value }: { value: string }) => {
-  const [date, time, ms] = value.split(/[T.Z]/);
+export const DateNode = ({ value }: { value: string | Date }) => {
+  let dateSplit: string[] = [];
+  if (value instanceof Date) {
+    try {
+      dateSplit = value.toISOString().split(/[T.Z]/);
+    } catch (error) {
+      console.error('Invalid date value provided to DateNode:', value);
+      dateSplit = ['Invalid date', '', ''];
+    }
+  } else {
+    dateSplit = value.split(/[T.Z]/);
+  }
+  const [date, time, ms] = dateSplit;
+
   const colors = useThemeColors();
   return (
     <span style={{ whiteSpace: 'nowrap', color: colors.date }}>
       {date}
-      <span style={{ opacity: 0.7 }}>T</span>
-      {time === '00:00:00' ? <span style={{ opacity: 0.7 }}>{time}</span> : time}
-      {ms === '000' ? <span style={{ opacity: 0.7 }}>.{ms}</span> : `.${ms}`}
-      <span style={{ opacity: 0.7 }}>Z</span>
+      { date !== 'Invalid date' && (
+        <>
+          <span style={{ opacity: 0.7 }}>T</span>
+          {time === '00:00:00' ? <span style={{ opacity: 0.7 }}>{time}</span> : time}
+          {ms === '000' ? <span style={{ opacity: 0.7 }}>.{ms}</span> : `.${ms}`}
+          <span style={{ opacity: 0.7 }}>Z</span>
+        </>
+      )}
     </span>
   );
```

Comment:
```jsx
  const colors = useThemeColors();
  if (!parsed) {
    return <span style={{ whiteSpace: 'nowrap', color: colors.date }}>Invalid date</span>;
  }

  const [date, time, ms] = parsed.toISOString().split(/[T.Z]/);
  return (
    <span style={{ whiteSpace: 'nowrap', color: colors.date }}>
      {date}
      <span style={{ opacity: 0.7 }}>T</span>
      {time === '00:00:00' ? <span style={{ opacity: 0.7 }}>{time}</span> : time}
      {ms === '000' ? <span style={{ opacity: 0.7 }}>.{ms}</span> : `.${ms}`}
      <span style={{ opacity: 0.7 }}>Z</span>
    </span>
  );
```

This is what I would do to decouple the date parsing from the render logic (splitting for purpose of displaying) and avoid the conditional in the template.